src/test/scala/uk/gov/nationalarchives/tdr/api/utils/FixedUUIDSource.scala

```diff
@@ -17,7 +17,16 @@ class FixedUUIDSource extends UUIDSource {
       UUID.fromString("0824d319-9c58-4b15-81eb-b71fb460fa36"),
       UUID.fromString("15eddae1-f319-4da1-b2da-cda3bd8f840a"),
       UUID.fromString("854b1c99-5998-4126-a209-6f1bcc38bf74")
-    )(idx)
+    )
+
+    // The static uuids above are used when we're testing the correct id is returned.
+    // Some methods call the uuid() method up to 20 times but we don't need to know what the uuid is as we're not testing for it.
+    // So we just return a random uuid.
+    if(idx >= uuids.size) {
```

Comment:
I already find this class quite confusing to work with. I'm worried that adding randomness will make it even harder!

Could we split this up, so each test can decide whether to use a fixed UUID source or a random UUID source?

Reply:
Yeah ok. makes sense. I've had to remove one of the assertions in the `createFile" should "create three files given correct arguments`  test because with the `RandomUUIDSource`, I can't tell which uuids are coming from the source but other than that, the tests work using a `RandomUUIDSource`

Reply:
Ignore that last comment. The `FixedUUIDSource` class lets us get the same uuids out each time it's called which allows us to test that IDs are being created correctly among other things. 
Now we're creating all the `FileMetadata` rows, the `FileService` class now calls `UUIDSource.uuid` more than 8 times which is why I added this random section in. But as you say, this isn't very neat. I did put in a `RandomUUIDSource` and used that in `FileServiceSpec` and this worked ok but `FileRouteSpec` is still using `FixedUUIDSource` (This is set in the test application.conf) and it's quite hard to change it to use the new RandomUUIDSource.

Anyway, long story short, I've added some more UUIDs to `FixedUUIDSource` so it doesn't run out when trying to create the `FileMetadata`.

Reply:
That's better for now, thanks. 👍 

I'd quite like to revisit FixedUUIDSource at some point because I think it makes it hard to work out which UUID is going to be used by which test, but I realise it's hard because of the way it's injected. I'll wait til I'm doing more work with these tests to try to see if I can come up with an option that's easier to understand.